src/main/webapp/app/entities/budget/budget-create.tsx

```diff
@@ -0,0 +1,164 @@
+
+export class EventBudgetCreate extends React.Component<IBudgetCreateProps> {
+  constructor(props) {
+    super(props);
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    if (nextProps.updateSuccess !== this.props.updateSuccess && nextProps.updateSuccess) {
+      this.handleClose();
+    }
+  }
+
+  componentDidMount() {
+    this.props.reset();
+  }
+
+  saveEntity = (event, errors, values) => {
+    if (errors.length === 0) {
+      const { budgetEntity } = this.props;
+      const entity = {
+        ...budgetEntity,
+        ...values
+      };
+
+      this.props.createEntity(entity);
+    }
+  };
+
+  handleClose = () => {
+    this.props.history.push(`/entity/event-budget/event/${this.props.match.params.eventId}`);
+  };
+
+  render() {
+    const { loading, updating, errorMessage } = this.props;
+    const { eventId } = this.props.match.params;
+
+    return (
+      <div>
+        <Row className="justify-content-center">
+          <Col md="8">
+            <h2 id="clubmanagementApp.eventBudget.home.createLabel" className="event-module-heading">
+              <Translate contentKey="clubmanagementApp.eventBudget.home.createLabel">Create Event Budget</Translate>
+            </h2>
+          </Col>
+        </Row>
+        <Row className="justify-content-center">
+          <Col md="8">
+            {loading ? (
+              <p>Loading...</p>
+            ) : (
+              <AvForm model={{}} onSubmit={this.saveEntity}>
+                <AvField hidden id="budget-eventId" type="string" className="form-control" name="eventId" value={eventId} />
+
+                <AvGroup>
+                  <Label id="amountLabel" for="budget-amount">
+                    <Translate contentKey="clubmanagementApp.eventBudget.amount">Amount</Translate>
+                  </Label>
+                  <AvField
+                    id="budget-amount"
+                    type="number"
+                    name="amount"
+                    validate={{
+                      required: { value: true, errorMessage: 'Please Enter an amount for this budget' },
+                      min: { value: 0, errorMessage: 'Amount cannot be less than 0' },
+                      pattern: { value: '^([0-9]*.?[0-9]{1,2})$', errorMessage: 'Please enter a valid amount with max of 2 decimal values' }
+                    }}
+                  />
+                </AvGroup>
+                <AvGroup>
+                  <Label id="nameLabel" for="budget-name">
+                    <Translate contentKey="clubmanagementApp.eventBudget.name">Name</Translate>
+                  </Label>
+                  <AvField
+                    id="budget-name"
+                    type="text"
+                    name="name"
+                    required
+                    validate={{
+                      required: { value: true, errorMessage: 'Please Enter a name for this budget' },
```

Comment:
e in Enter should be lower capital.

Reply:
resolved in 948ee18a02eeba1d00248750fb41673c50daa860